### ISSUE TITLE
LDAP client unit tests and samples should not depend on ApacheDSContainer

### ISSUE THREAD

**Issue by jzheaux** [spring-projects/spring-security#6002]
ApacheDS does not have a GA release that is compatible with JDK 11.

Several `spring-security-ldap` unit tests and LDAP samples use `ApacheDSContainer` for convenience in setup. However, `UnboundIdContainer` is compatible with JDK 11, meaning that their dependency on `ApacheDSContainer` is not necessary.

**Comment by jzheaux** [comment]
Related to #5860

**Comment by LukeButtersFunnelback** [comment]
ApacheDS havn't released a GA in 8 years they all seem to be mile stone releases. Would it be worth updating ApacheDSContainer to be compatible with one newer version of apache DS?

**Comment by jzheaux** [comment]
@LukeButters I don't think so since, as you say, they aren't GA releases. ApacheDS's `2.x` milestones are a bit of a moving target, making it hard to pick one that we can be confident in. But, really, it'd be better to stick with a GA release either way.

Also, point releases following `1.5.5` have breaking changes that make it tricky to upgrade along the `1.5.x` train.

**Comment by LukeButters** [comment]
I guess if someone could get a sample ApacheDSContainer working for a `2.x` version it might be easier for other devs to get working different versions. I guess I want that because I tried and gave up, but I suspect other devs will get into the same issue when trying to upgrade to jdk11

**Comment by rwinch** [comment]
@LukeButters Thanks for the feedback. I think that would be a good idea. Would you submit a PR marking ApacheDSContainer as deprecated and steering users to UnboundId support?

**Comment by LukeButters** [comment]
I made a PR https://github.com/spring-projects/spring-security/pull/6376 hopefully that is how you want it.
